packages/components/src/SubHeaderBar/InputTitleSubHeader/InputTitleSubHeader.component.js

```diff
@@ -0,0 +1,162 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { Action } from '../../Actions';
+import theme from './InputTitleSubHeader.scss';
+import { getDefaultTranslate } from '../../translate';
+import Icon from '../../Icon';
+
+function DetailsTitle({ title, subTitle, onEdit, t }) {
+	const modifyLabel = t('MODIFY_TOOLTIP', { defaultValue: 'Modify' });
+	return (
+		<div className={classNames(theme['tc-subheader-details-text'], 'tc-subheader-details-text')}>
+			<div
+				className={classNames(
+					theme['tc-subheader-details-text-title'],
+					'tc-subheader-details-text-title',
+				)}
+			>
+				<h1
+					className={classNames(
+						theme['tc-subheader-details-text-title-wording'],
+						'tc-subheader-details-text-title-wording',
+					)}
+				>
+					{title}
+				</h1>
+				{onEdit && (
+					<Action
+						name="action-edit-title"
+						label={modifyLabel}
+						icon="talend-pencil"
+						onClick={onEdit}
+						bsStyle="link"
+						className={classNames(
+							theme['tc-subheader-details-text-title-pencil'],
+							'tc-subheader-details-text-title-pencil',
+						)}
+						hideLabel
+					/>
+				)}
+			</div>
+			{subTitle && (
+				<small
+					className={classNames(
+						theme['tc-subheader-details-text-subtitle'],
+						'tc-subheader-details-text-subtitle',
+					)}
+				>
+					{subTitle}
+				</small>
+			)}
+		</div>
+	);
+}
+
+DetailsTitle.propTypes = {
+	title: PropTypes.string.isRequired,
+	subTitle: PropTypes.string,
+	onEdit: PropTypes.func,
+	t: PropTypes.func,
+};
+
+DetailsTitle.defaultProps = {
+	t: getDefaultTranslate,
+};
+
+function onFocus(event) {
+	event.target.selectionStart = 0; // eslint-disable-line no-param-reassign
+	event.target.selectionEnd = event.target.value.length; // eslint-disable-line no-param-reassign
```

Comment:
no please use ref callback 
onRef(ref) {
  if (ref) { ref.select(); ref.focus(); }
}
and remove autoFocus